modules/exploits/linux/http/apache_ofbiz_deserialization.rb

```diff
@@ -76,21 +81,27 @@ def initialize(info = {})
 
     register_options([
       Opt::RPORT(8443),
-      OptString.new('TARGETURI', [true, 'Base path', '/'])
+      OptString.new('TARGETURI', [true, 'Base path', '/']),
+      OptBool.new('AUTHBYPASS', [true, 'Utilize CVE-2023-51467', true])
     ])
   end
 
   def check
     # Send an empty serialized object
     res = send_request_xmlrpc('')
 
-    unless res
-      return CheckCode::Unknown('Target did not respond to check.')
-    end
+    return CheckCode::Unknown('Target did not respond to check.') unless res
 
-    if res.body.include?('Failed to read result object: null')
-      return CheckCode::Vulnerable('Target can deserialize arbitrary data.')
-    end
+    return CheckCode::Vulnerable('Target can deserialize arbitrary data.') if res.body.include?('Failed to read result object: null')
+
+    # newer versions respond w/o a content length, so try just a gest
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, '/webtools/control/xmlrpc')
+    )
+
+    return CheckCode::Unknown('Target did not respond to check.') unless res
+
+    return CheckCode::Detected('Apache OFBiz detected') if res.body.include?('Apache OFBiz.')
 
     CheckCode::Safe('Target cannot deserialize arbitrary data.')
```

Comment:
```suggestion
    return CheckCode::Safe('Target cannot deserialize arbitrary data.')
```

To be consistent with the previous lines.

Reply:
last statement of function, returned anyways so no need to return